### ISSUE TITLE
fields.remove (and non-closure pattern) always removes last element in FieldArray

### ISSUE THREAD

**Issue by markau** [erikras/redux-form#2291]
I am using `onClick={() => fields.remove(index)}` from the [docs](http://redux-form.com/6.3.1/examples/fieldArrays/), and have also tried the non-closure workaround outlined in issue #1128, and the example provided in release 1.2.1.

In each case, no matter which `FieldArray` element I click _'close'_ on, it always removes the last element in the `FieldArray`.

If I add `console.log(index)` to the `onClick` method, I see that the appropriate index number is logged to the console, but still the last element in `FieldArray` is what gets removed. 😕  Indeed, I can call `fields.remove(0)` and still the last element gets popped.

Usually I'd assume I've done something wrong - and that remains a strong possibility - but, I have followed the examples closely and it's quite straightforward.

TL;DR; `fields.remove(index)` is behaving like `fields.pop()`.

I am using `"redux-form": "6.3.2"`

**Comment by markau** [comment]
Some further debugging / tearing hair out led me to realise that the issue is due to the components I am using being the [react-rte](https://github.com/sstur/react-rte) component via [this method](https://gist.github.com/erikras/5ca8dc618bae5dbc8faf7d2324585d01).

Unsurprisingly, this isn't an issue with redux-form after all, so I'll close the issue.

**Comment by MartinCerny-awin** [comment]
I have the same problem. Have you solved your issue @markau? I can't find out why is it happening.

**Comment by markau** [comment]
I can only offer a workaround as it's pushing the limit of my understanding. See [this comment in the gist](https://gist.github.com/erikras/5ca8dc618bae5dbc8faf7d2324585d01#gistcomment-1951377): if I define the stateless `Field` component outside the `render` function (against the advice in the docs) then it forces the `Fields` to update upon any change of state - in this case, removing an item from the `FieldArray`.  I would be very keen to hear of a better approach! 

Let me know if you want me to paste a code sample and I'll do that when I return to the office.

**Comment by ddcech** [comment]
@markau could you please provide an example how you got this to work. This bug is honestly driving me nuts. Thanks in advance.

**Comment by markau** [comment]
Hey @ddcech, although I am sorry to hear you are having issues, it is good to get further confirmation of the issue!

I added [a comment](https://gist.github.com/erikras/5ca8dc618bae5dbc8faf7d2324585d01#gistcomment-1951377) to the gist that shows my workaround. (Whilst this is in relation to the RTE component, you could substitute any component.)

Does this help?  Likewise any further info you can share is helpful in getting to the bottom of it.

**Comment by markau** [comment]
At least four people affected, I think that's worth reopening the issue?

**Comment by gustavohenke** [comment]
It will be awesome if anyone can provide a working example. https://jsfiddle.net/gustavohenke/48814zsq/

**Comment by ddcech** [comment]
Thanks! @markau, your solution works for me. I am using the rc-time-picker component.

**Comment by markau** [comment]
@gustavohenke I have [forked your fiddle and setup the example](https://jsfiddle.net/Lgpa2qz8/) - can't reproduce :cry: - will try again a bit later. I have completely confused myself.

**Comment by markau** [comment]
hey @gustavohenke, this is proving difficult to replicate in a fiddle. I can say, however, that the problem is where the `renderField` (using terminology from the members/hobbies example) includes a component which uses `componentDidMount `to set the `state`.

The complex component (versus the simple `<input type="text" {...input} />`) from the members example) does not have its props 'refreshed' when the `fields `array changes. Similar to [this issue on SO](http://stackoverflow.com/questions/30028086/component-not-updating-on-props-update).

So, not a bug in redux-form. I'll close it again 😆. 

If you know how to force the props to refresh in the component upon changes to the `fields`, I would be absolutely grateful for any hint whatsoever.

**Comment by samuelg0rd0n** [comment]
@markau Could you please explain what you mean by "the solution is to use componentWillReceiveProps to refresh the props in the component."? I don't understand what component you are talking about, what props to refresh and how.
